tests/api_tests/nodes/test_views.py

```diff
@@ -689,6 +697,40 @@ def test_node_update_invalid_data(self):
         assert_equal(res.status_code, 400)
         assert_equal(res.json['errors'][0]['detail'], "Malformed request.")
 
+    @assert_not_logs(NodeLog.UPDATED_FIELDS, 'private_project')
+    def test_cannot_make_project_public_if_non_contributor(self):
+        non_contrib = AuthUserFactory()
+        res = self.app.patch_json(
+            self.private_url,
+            make_node_payload(self.private_project, {'public': True}),
+            auth=non_contrib.auth, expect_errors=True
+        )
+        assert_equal(res.status_code, 403)
+
+    def test_cannot_make_project_public_if_non_admin_contributor(self):
+        non_admin = AuthUserFactory()
+        self.private_project.add_contributor(non_admin, auth=Auth(self.private_project.creator))
```

Comment:
What is a non-admin contributor? Is it explicitly a write contributor, or does it mock something that goes a little deeper into our permissions and generically give all permissions except for admin?